src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/InProcess/EditorInProcess_Caret.cs

```diff
@@ -27,6 +27,33 @@ public async Task MoveCaretAsync(int position, CancellationToken cancellationTok
             view.Caret.MoveTo(point);
         }
 
+        public async Task WaitForCaretMoveAsync(CancellationToken cancellationToken)
+        {
+            var view = await GetActiveTextViewAsync(cancellationToken);
+
+            using var semaphore = new SemaphoreSlim(1);
+            await semaphore.WaitAsync(cancellationToken);
+
+            view.Caret.PositionChanged += Caret_PositionChanged;
```

Comment:
I discover things by looking at existing code, poking around in intellisense, and poking around in roslyn, and from that point of view it was fine.

Worth noting, I could have done this with existing helpers and polling, rather than events, so duplication is easy enough to remove.